corehq/pillows/synclog.py

```diff
@@ -100,23 +100,24 @@ def process_change(self, change):
             user = CouchUser.get_by_user_id(user_id)
             if not user:
                 return
-            mark_last_synclog(domain, user, app_id, build_id, sync_date, device_id)
 
+            device_app_meta = None
+            if device_id and app_id:
+                device_app_meta = DeviceAppMeta(app_id=app_id, build_id=build_id, last_sync=sync_date)
+            mark_last_synclog(domain, user, app_id, build_id, sync_date, device_id, device_app_meta)
 
-def mark_last_synclog(domain, user, app_id, build_id, sync_date, device_id, save_user=True):
+
+def mark_last_synclog(domain, user, app_id, build_id, sync_date, device_id, device_app_meta, build_profile_id=None, save_user=True):
```

Comment:
not clear about the changes to synclog, why is this being done here?

Reply:
What's happening in the heartbeat view [above](https://github.com/dimagi/commcare-hq/pull/26305/files#diff-291dfd3802bf0daa51e598788d5ca13bL352) is pretty much same as what's happening inside `mark_last_synclog`, so I am modifying it to make this more DRY

Reply:
i am not sure about other fields but `build_profile_id` is only present in heartbeat requests and not in sync log, so will this clear out the value set for it via heartbeat?

Reply:
Each action just updates the relevant fields doesn't override other fields. See `add_sync` method [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/models.py#L2850) which doesn't override that field.

Reply:
I am not sure if that link took me to where you expected me to see,
it was on the `ON CONFLICT (domain, user_id, app_id)` line in the sql,
is that what you were pointing to?

the github links with master as reference would keep changing, i find it useful to have them based on a commit which would not change.

Reply:
Yeah.